### PR TITLE
Updating 1.7.x to support SimpleSAMLphp 2.0.x, new directory structure.

### PR DESCRIPTION
**Pull request by dmundra**
Includes fixes from https://github.com/drupalauth/simplesamlphp-module-drupalauth/pull/44/files.

The changes make the Drupal 7 version work with SimpleSAMLphp 2.0.x. The Drupal module will need to be patched with the patch from https://www.drupal.org/project/drupalauth4ssp/issues/3439480#comment-15542623.

### CODE REVIEW

File: src/Auth/Source/External.php [hunk 1]
```diff
@@ -225,13 +230,13 @@ private function getUser() {
...
-      $current_dir = getcwd();
+      $a = getcwd();
       chdir(DRUPAL_ROOT);
 
       // load the user object from Drupal
-      $drupaluser = user_load($drupaluid, TRUE);
+      $drupaluser = user_load($drupaluid);
 
-      chdir($current_dir);
+      chdir($a);
```
Comment (RoSk0) [inline src/Auth/Source/External.php:239]: Those changes are not related to SSP 2.0.x support. Please revert them

File: src/Auth/Source/External.php [hunk 2]
```diff
@@ -296,16 +301,13 @@ private function getUser() {
...
+            catch (\Exception $e) {
               watchdog_exception('simplesaml', $e);
             }
           }
         }
       }
-
-      chdir(DRUPAL_ROOT);
       drupal_alter('drupalauth_attributes', $attributes, $drupaluser);
-      chdir($current_dir);
```
Comment (RoSk0) [inline src/Auth/Source/External.php:308]: Those should be restored.

File: src/Auth/Source/UserPass.php [hunk 3]
```diff
@@ -73,7 +77,7 @@
  * @package drupalauth
  * @version $Id$
  */
-class sspmod_drupalauth_Auth_Source_UserPass extends sspmod_core_Auth_UserPassBase {
+class UserPass extends \SimpleSAML\Module\core\Auth\UserPassBase {
```
Comment (RoSk0) [inline src/Auth/Source/UserPass.php:80]: Please reference `UserPassBase` using `use` statement.

**Review by RoSk0 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

However there are couple things that must happen before this can be merged. First of all you will need to rebase your branch onto newly created `2.7.x` (it includes PHP 7.2 compatibility fix merged) and also change pull request target to that branch. Plus address all review comments.

**Comment by dmundra** [comment]
Thank you @RoSk0, I will make the changes

File: src/Auth/Source/External.php (hunk 1 above)
Comment (dmundra) [reply src/Auth/Source/External.php]: I reset the changes.

File: src/Auth/Source/External.php (hunk 2 above)
Comment (dmundra) [reply src/Auth/Source/External.php]: Yup good catch. Restored.

File: src/Auth/Source/UserPass.php (hunk 3 above)
Comment (dmundra) [reply src/Auth/Source/UserPass.php]: Fixed

**Comment by dmundra** [comment]
@RoSk0 made the changes, rebased, and changed the target branch. How is it looking now?
